cmd/virt-apiservice-register/virt-apiservice-register.go

```diff
@@ -0,0 +1,737 @@
+/*
+Copyright 2017 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at:
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file implements a tool that simplifies registration of API services.
+
+package main
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/base64"
+	"encoding/pem"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/golang/glog"
+	certv1 "k8s.io/api/certificates/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Values of the command line flags:
+//
+var (
+	apiGroup           string
+	apiGroupPriority   int
+	apiVersion         string
+	apiVersionPriority int
+	dnsDomain          string
+	kubeConfig         string
+	namespaceName      string
+	secretName         string
+	serviceName        string
+	servicePort        int
+	targetPort         int
+	targetSelector     string
+)
+
+func main() {
+	// Define the command line options:
+	flag.StringVar(
```

Comment:
This is just a matter of personal preference. I like to see the flags declarations on a single line. I don't really care about this very much though.